main.py

```diff
@@ -2,43 +2,69 @@
 
 # Press Shift+F10 to execute it or replace it with your code.
 # Press Double Shift to search everywhere for classes, files, tool windows, actions, and settings.
+"""
+Challenge:
+#. Implement ``Product`` class
+#. Each ``Product`` instance should implement properties:
+    * ``name`` - a product's name, like apple, cheese etc.
+    * ``price`` - a price for a single unit
+#. ``Product`` instance should have a behavior of calculating total
+   price for a specified quantity of goods
+#. Implement ``ShoppingCart`` class
+#. ``ShoppingCart`` instance should combine products instances and
+    corresponding purchased quantities.
+#. ``ShoppingCart`` instance should implement a method to calculate
+   the total price of entire cart.
+"""
+
 class Product:
-    def __init__(self, name: str, price: int, amount: int):
+    """class Product imlementation
+    * ``name`` - a product's name, like apple, cheese etc.
+    * ``price`` - a price for a single unit
+    """
+    def __init__(self, name: str, price: int):
         self.name = name
         self.price = price
-        self.amount = amount
 
-    def total_price(self):
-        return self.amount * self.price
 
+    def calculating_total_price_for_product(self, quantity_of_goods):
```

Comment:
Це дуже довга не зрозуміла занадто деталізована назва що використовує зміїний стиль написання для того щоб люди яка її читає було одразу зрозуміло з самої назви що цей метод відповідає за підрахунок кошториса та використовує значення кількості товарів :-)

Де саме, ти загубив суть, поки читав оте що зверху?
Давай будемо лаконічними - залиш довгі пояснення для docstring.
```suggestion
    def get_total(self, quantity):
```